### PR TITLE
Twister: fix terminology

### PR DESCRIPTION
**Pull request by nashif**
An attempt to overhaul twister naming conventions and get it to closer to the structure of tests we have now. This shall avoid confusion and eliminates legacy naming of various classes and how we process results.

Still work in progress.

### CODE REVIEW

File: scripts/twister [hunk 1]
```diff
@@ -556,12 +559,12 @@ structure in the main Zephyr tree: boards/<arch>/<board_name>/""")
     parser.add_argument(
         "-S", "--enable-slow", action="store_true",
         help="Execute time-consuming test cases that have been marked "
-             "as 'slow' in testcase.yaml. Normally these are only built.")
+             "as 'slow' in tests.yaml. Normally these are only built.")
```
Comment (hakehuang) [inline scripts/twister:562]: I think this should be testcase.yaml not tests.yaml.

File: scripts/twister [hunk 2]
```diff
@@ -556,12 +559,12 @@ structure in the main Zephyr tree: boards/<arch>/<board_name>/""")
...
 
     parser.add_argument(
         "-K", "--force-platform", action="store_true",
         help="""Force testing on selected platforms,
-        even if they are excluded in the test configuration (testcase.yaml)"""
+        even if they are excluded in the test configuration (tests.yaml)"""
```
Comment (hakehuang) [inline scripts/twister:567]: I think this should be testcase.yaml not tests.yaml.

**Review by hakehuang — APPROVED** [review]
I guess there are two miss replacements from 'testcase.yaml` to 1tests.yaml`, comments in line. others looks good to me

**Review by LeiW000 — APPROVED** [review]

**Comment by PerMac** [comment]
@hakehuang I believe these are not missed replacements. One of the reasons behind this PR was issues with the naming convention, in particular, "test case" was problematic as it was used to name both test cases defined in .c and test suites (collections of test cases).

**Comment by PerMac** [comment]
@nashif At the first glance, the reports look fine. I started reviewing the code a while ago but decided to wait until you finish the pr. I will also ask you if some clean ups in the commits can be done (e.g. there are lines that are changed multiple times in multiple commits, hence reviewing on the commit base is almost impossible). 
I think the structure of `instance` objects in the code should be modified: now you have `self.instance.scenario.suite.(instances/name)` i.e. `suite` is sitting inside `scenario`. `suite` also has `scenarios` field, so the relation looks a bit convoluted. In fact, when debugging in PyCharm it looks like an infinite loop: `self.instance.scenario.suite.instances[0].scenario.suite.instances ...`
From my understanding `suite` is at a higher level of tests' organization than `scenario`.

**Comment by hakehuang** [comment]
> @hakehuang I believe these are not missed replacements. One of the reasons behind this PR was issues with the naming convention, in particular, "test case" was problematic as it was used to name both test cases defined in .c and test suites (collections of test cases).

@PerMac , we only have testcase.yaml not test.yaml.

**Comment by PerMac** [comment]
@hakehuang For now, yes. But the point is to switch testcase.yaml -> tests.yaml. This PR adds support for this: 
https://github.com/zephyrproject-rtos/zephyr/pull/32541/commits/c0405b18d674a67c947c97519a4cec8ae4a58e91
So the default will be tests.yaml but testcase.yaml will also be checked for legacy reason

**Comment by nashif** [comment]
> I will also ask you if some clean ups in the commits can be done (e.g. there are lines that are changed multiple times in multiple commits, hence reviewing on the commit base is almost impossible).

at some point many of those will be squashed, I suggest to look at the change as a whole right now, since this change is about one thing only  and not trying to introduce new features..

**Pull request closed without merging**